src/test/java/com/marklogic/spark/writer/StreamRowsTest.java

```diff
@@ -26,39 +30,63 @@
+        newDefaultStreamWriter(newSparkSession(), "src/test/resources/inputForStream", SCHEMA, "csv")
             .outputMode("append")
-            .option(Options.WRITE_COLLECTIONS, "my-test-data")
+            .option(Options.WRITE_URI_TEMPLATE, "/hogwarts/{Name}.json")
+            .option(Options.WRITE_COLLECTIONS, collection)
             // checkpointLocation needs to be a new location everytime job is run else docs are not written.
             .option("checkpointLocation", tempDir.toFile().getAbsolutePath()).start().processAllAvailable();
-        final int expectedCollectionSize = 9;
-        String uri = getUrisInCollection("my-test-data", expectedCollectionSize).get(0);
-        JsonNode doc = readJsonDocument(uri);
-        assertTrue(doc.has("Name"));
-        assertTrue(doc.has("House"));
+
+        assertCollectionSize(collection, 9);
+        JsonNode doc = readJsonDocument("/hogwarts/Hermione Granger.json");
+        assertEquals("Gryffindor", doc.get("House").asText());
+        assertEquals("Hermione Granger", doc.get("Name").asText());
     }
 
     @Test
-    void inValidTransform(@TempDir Path tempDir) {
-        try{
-            newDefaultStreamWriter(newSparkSession(), "src/test/resources/inputForStream", struct, "csv")
-                .outputMode("append")
-                .option(Options.WRITE_COLLECTIONS, "my-test-data")
-                .option(Options.WRITE_TRANSFORM_NAME, "this-doesnt-exist")
-                .option("checkpointLocation", tempDir.toFile().getAbsolutePath()) // checkpointLocation needs to be a new location everytime job is run else docs are not written.
-                .start().processAllAvailable();
-        } catch (Exception ex){
-            assertTrue(ex.getMessage().contains("XDMP-MODNOTFOUND: (err:XQST0059) Module /marklogic.rest.transform/this-doesnt-exist/assets/transform.xqy not found"));
-        }
+    void invalidTransform(@TempDir Path tempDir) throws Exception {
+        StreamingQuery query = newDefaultStreamWriter(newSparkSession(), "src/test/resources/inputForStream", SCHEMA, "csv")
+            .outputMode("append")
+            .option(Options.WRITE_COLLECTIONS, "my-test-data")
+            .option(Options.WRITE_TRANSFORM_NAME, "this-doesnt-exist")
+            .option("checkpointLocation", tempDir.toFile().getAbsolutePath())
+            .start();
+
+        // No error will be thrown yet as the streaming occurs in the background, though we'll likely see errors
+        // from the background thread. Calling processAllAvailable should force an error to occur.
+
+        StreamingQueryException ex = assertThrows(StreamingQueryException.class, () -> query.processAllAvailable());
+        assertTrue(ex.getMessage().contains("Extension this-doesnt-exist or a dependency does not exist"),
+            "Unexpected error: " + ex.getCause());
     }
+
+    private DataStreamWriter newDefaultStreamWriter(SparkSession session, String path, StructType struct, String format) {
```

Comment:
I moved this here, as it's only used by this test. If we have a separate test class in the future that needs this, it can move up to the parent class. But until then, having it here makes this test easier to read and understand.